backend/owa.ts

```diff
@@ -25,31 +32,53 @@ export async function fetchSessionData(partition: string, url: string, interacti
           return;
         }
         finished = true;
-        session.cookies.removeListener('changed', checkCanary);
+        session.cookies.removeListener('changed', onCookie);
         resolve(data);
         if (!popup.isDestroyed()) {
           popup.destroy();
         }
       };
       let checkLoginFinished = async function() {
         try {
+          url = urlObj.href;
           response = await session.fetch(url + 'sessiondata.ashx', { method: 'POST' });
-          finish(await response.json());
+          let json = await response.json();
+          if (isHotmail) {
+            json.owaURL = url;
+          }
+          finish(json);
         } catch (ex) {
         }
       }
-      let checkCanary = function(_event, cookie, _cause, removed) {
-        if (!removed && cookie.name == 'X-OWA-CANARY') {
-          checkLoginFinished();
+      let onCookie = async function(_event, cookie, _cause, removed) {
+        if (removed) {
+          return;
+        }
+        // For Hotmail, check the path to the CANARY cookie.
+        if (cookie.domain == kHotmailServer &&
+          cookie.name == kCookieName &&
+          cookie.path.startsWith("/owa/")) {
+          // Hotmail also sets cookies for /owa/0/, /mail/0/, /calendar/0/ etc.,
+          // but we can use only the /owa/0/ cookie.
+          // We also need to use that URL for the service request.
+          // This needs to happen before CheckLoginFinished().
+          urlObj.hostname = cookie.domain;
+          urlObj.pathname = cookie.path;
+          isHotmail = true;
+        }
+        // If we receive the canary cookie, check whether we're logged in
+        if (cookie.name == kCookieName &&
+            cookie.domain == urlObj.hostname) {
```

Comment:
Do we need this domain check condition? It wasn't there before.

Reply:
I think we need it just in case we don't get the correct cookie for personal accounts yet i.e. the cookie should be for `outlook.live.com` instead of `outlook.office.com`.

Reply:
It could also be because the cookie notifications in a browser extension aren't filtered by partition like they are here. (So in fact the Owl code should start checking the `cookieStoreId` of the cookie as well as or instead of the domain.)